server/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -824,14 +835,18 @@ public static RepositoryData snapshotsFromXContent(XContentParser parser, long g
                     indexMetaIdentifiers = parser.mapStrings();
                 }
                 case MIN_VERSION -> {
-                    IndexVersion version = parseIndexVersion(parser.nextToken(), parser);
-
-                    assert SnapshotsService.useShardGenerations(version);
-                    if (version.after(IndexVersion.current())) {
-                        throw new IllegalStateException(
-                            "this snapshot repository format requires Elasticsearch version [" + version + "] or later"
+                    final var token = parser.nextToken();
+                    XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_STRING, token, parser);
```

Comment:
Can we add a comment to help the next person who changes the `min_version`?

Reply:
Yeah I think this area needs more thought, `IndexVersion` isn't really the right thing to report to users in the error message. I added comments in 39ee760b134be74da235e96e806053c43daf9c2a.